java/flight/src/main/java/org/apache/arrow/flight/FlightService.java

```diff
@@ -191,17 +191,17 @@ public void completed() {
             StreamPipe.wrap(responseObserver, PutResult::toProtocol)).run();
         responseObserver.onCompleted();
       } catch (Exception ex) {
-        logger.error("Failed to process custom put.", ex);
         responseObserver.onError(ex);
-        // The client may have terminated, so the exception here is effectively swallowed.
         // Log the error as well so -something- makes it to the developer.
         logger.error("Exception handling DoPut", ex);
-      }
-      try {
-        fs.close();
-      } catch (Exception e) {
-        logger.error("Exception closing Flight stream", e);
-        throw new RuntimeException(e);
+        throw new RuntimeException(ex);
+      } finally {
+        try {
+          fs.close();
+        } catch (Exception e) {
+          logger.error("Exception closing Flight stream", e);
+          throw new RuntimeException(e);
```

Comment:
I don't think we want to throw here, it would override the exception from the inner block which seems to the more important of the two?

Reply:
This is an interesting question: 

If we get an exception in the main logic, and before leaving the current stack, we enter the finally block. In the finally block, we get another exception.

What will happen here? Should be see both exceptions, or the latter will override the former one?

Reply:
I conduct a small experiment, and the result shows that the exception in the finally block will override the exception in the main logic. You are right @emkornfield 

I think it is difficult to say which exception is more important. The exception in the main logic is more interesting to the user. However, the exception in the finally block is related to resource leak. In addition, it causes problems even if the main logic finishes successfully.

Reply:
I think the main one is probably better to keep, generally cleanup exceptions are rarer and likely less interesting.

Reply:
but per above, I don't think we should propagate for now.